lib/puppet/network/http_pool.rb

```diff
@@ -73,16 +74,18 @@ def self.http_ssl_instance(host, port, verifier = Puppet::SSL::Validator.default
   # @api public
   #
   def self.connection(host, port, use_ssl: true, ssl_context: nil)
+    # When running in a JRuby master environment the Server manages our
+    # SSLContext for us.
     if use_ssl
-      unless ssl_context
+      if !ssl_context && !Puppet::Util::Platform.jruby?
```

Comment:
If we allow a nil ssl_context then other code needs to account for it possibly being nil like the `Verifier` below. Could we instead have puppetserver create a dummy/empty context and push that on the stack? Then we wouldn't need to make this conditional on jruby.

Reply:
In a separate thread about trust anchors we talked about the SSLContext being a thing that JRuby interrogated to find out what certs to load for its http client. With the caveat that the SSLContext it interrogated mustn't load OpenSSL for FIPS in JRuby compatibility.

This current work seems like it would be superseded by that other SSLContext work. eg, this is difficult because the Server's http client class doesn't have the same constructor API as Puppet's Ruby client wrt SSLContext & Verifier (what this method is concerned with). Maybe instead of creating an empty context, it'd be better to create that non-OpenSSL context that Server can interrogate for certs, then update the Server's http client class to honor that context in its constructor?

Or, going in the opposite direction, could we move these checks that happen in this method into the client class (it becomes the responsibility of the Ruby or JRuby client classes to find and validate their ssl contexts)?

I might be way off base on the connection between these two issues....

Reply:
tl;dr was that I was off base